cmd/main.go

```diff
@@ -76,28 +75,28 @@ func main() {
 		configFile               string
 	)
 
-	flag.StringVar(&metricsAddr, "metrics-bind-address", ":8443", "The address the metric endpoint binds to.")
-	flag.StringVar(&probeAddr, "health-probe-bind-address", ":8081", "The address the probe endpoint binds to.")
-	flag.Float64Var(&qps, "kube-api-qps", 500, "Maximum QPS to use while talking with Kubernetes API")
-	flag.IntVar(&burst, "kube-api-burst", 500, "Maximum burst for throttle while talking with Kubernetes API")
+	flag.StringVar(&metricsAddr, "metrics-bind-address", ":8443", "DEPRECATED(please pass configuration file via --config flag): The address the metric endpoint binds to.")
```

Comment:
I'm thinking whether we're too aggressive to deprecate all the flags, should we leave some basic flags for easy configuration, like `metricsAddr` `probeAddr` `qps` `burst` `enableLeaderElection`. I need some feedbacks.

Reply:
@kerthcet thank you for your review. I think it makes sense. 

* `kube-api-qps`
* `kube-api-burst`
* `leader-elect`
will  also be supported as flag. I agree with you that if someone wants to explicitly enable leader election, it would be overkill to force using a configuration file and it would be easier by simply passing `leader-elect` flag.

Deprecated ones by this PR;

* `metrics-bind-address`
* `health-probe-bind-address`
* `leader-elect-lease-duration`
* `leader-elect-renew-deadline`
* `leader-elect-retry-period`
* `leader-elect-resource-lock`
* `leader-elect-resource-name`
These configurations seem to fall in to fine-grained tuning category and better to be only represented in configuration file rather than a flag?.